File: progSys/gestionDD/C/exo3b.c
#include <stdio.h>
#include <stdlib.h>
#include <fcntl.h>
#include <unistd.h>
#include <sys/stat.h>

int main()
{
    struct stat st;

    if (stat("tableau.bin", &st) == -1)
    {
        perror("stat");
        return 1;
    }

    int n = st.st_size / sizeof(int);

    int *tab = malloc(n * sizeof(int));

    int fd = open("tableau.bin", O_RDONLY);

    if (fd == -1)
    {
        perror("open");
        return 1;
    }

    read(fd, tab, n * sizeof(int));

    close(fd);

    printf("Contenu du tableau :\n");

    for (int i = 0; i < n; i++)
    {
        printf("%d ", tab[i]);
    }

    printf("\n");

    free(tab);

    return 0;
}

File: progSys/gestionDD/python/exo3b.py
import array
import os

taille = os.stat("tableau.bin").st_size
tab = array.array('i')
with open("tableau.bin", "rb") as f:
    tab.fromfile(f, taille // tab.itemsize) 
print("Contenu du tableau :")
for val in tab:
    print(val, end=' ')
print()